File: 백준/Silver/1149. RGB거리/RGB거리.py
n = int(input())
cost = [list(map(int,input().split())) for _ in range(n)]

for i in range(1,n):
    # 빨 -> 전 집의 초,파를 선택했을 때의 최소값과 합한 것으로 결정
    cost[i][0] = min(cost[i-1][1],cost[i-1][2]) + cost[i][0]

    # 초
    cost[i][1] = min(cost[i-1][0],cost[i-1][2]) + cost[i][1]

    # 파
    cost[i][2] = min(cost[i - 1][0], cost[i - 1][1]) + cost[i][2]

print(min(cost[n-1]))

File: 백준/Silver/1149. RGB거리/RGB거리.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
// '/dev/stdin'

const n = +input[0];
const cost = [];

for(let i = 0; i < n; i++){
    let [r,g,b] = input[i+1].split(' ').map(Number);
    cost.push([r,g,b]);
}

let res = 1000;

for(let i = 1; i < n; i++){
    cost[i][0] += Math.min(cost[i-1][1], cost[i-1][2]);
    cost[i][1] += Math.min(cost[i-1][0], cost[i-1][2]);
    cost[i][2] += Math.min(cost[i-1][0], cost[i-1][1]);
}
console.log(Math.min(cost[n-1][0],cost[n-1][1],cost[n-1][2]));